hera_qm/xrfi.py

```diff
@@ -1975,13 +1984,50 @@ def xrfi_run(ocalfits_file=None, acalfits_file=None, model_file=None, data_file=
                 'abscal_chi_sq_renormed_metrics2.h5': uvf_az2, 'abscal_chi_sq_flags2.h5': uvf_azf2,
                 'combined_metrics2.h5': uvf_metrics2, 'combined_flags2.h5': uvf_fws2,
                 'flags2.h5': uvf_combined2}
-    basename = qm_utils.strip_extension(os.path.basename(output_prefix))
-    for ext, uvf in uvf_dict.items():
-        if uvf is not None:
-            outfile = '.'.join([basename, ext])
-            outpath = os.path.join(dirname, outfile)
-            uvf.write(outpath, clobber=clobber)
 
+    # Determine the actual files to store
+    # We will drop kt_size / (integrations per file) files at the start and
+    # end to avoid edge effects from the convolution kernel.
+    # If this chunk includes the start or end of the night, we will write
+    # output files for those, but flag everything.
+
+    # Read metadata from first file to get integrations per file.
+    file_lists = [ocalfits_files, acalfits_files, data_files, model_files]
+    if data_files is not None:
+        dtypet = 'uvdata'
+        uvlist = data_files
+    elif model_files is not None:
+        dtypet = 'uvdata'
+        uvlist = model_files
+    elif ocalfits_files is not None:
+        dtypet = 'uvcal'
+        uvlist = ocalfits_files
+    elif acalfits_files is not None:
+        dtypet = 'uvcal'
+        uvlist = acalfits_files
+    if dtypet == 'uvdata':
+        uvtemp = UVData()
+        uvtemp.read(uvlist[0], read_data=False)
+    elif dtypet == 'uvcal':
+        uvtemp = UVCal()
+        uvtemp.read_calfits(uvlist[0])
+
+    # Loop through the files to output, storing all the different data products.
```

Comment:
It looks like your dropped the feature where you only output files for central files. @adampbeardsley's idea here was to try to avoid edge-effects by deriving flags for a set of files using a slightly larger set of files and then not writing down the flags from the results from the edge files. The new OPM behavior lets you do this kind of chunking where the chunk size is larger than the stride length, allowing for flagging on overlapping sets of files. 

I think this behavior should be on by default when you have multiple files, but I'm open to it being default false. Regardless, we need it to be triggerable from the command line with a boolean flag. 

See https://github.com/HERA-Team/hera_qm/blob/5313e1f41bc256ad5e9ca163039515f5c24956ae/hera_qm/xrfi.py#L2226-L2266
for the implementation

Reply:
I ended up leaving this out because it broke the day_thresholding unittest and it wasn't clear to me why it existed. Now that you've explained it, I see it's use. I can add it back in and disable it on day thresholding.

Reply:
Thanks! Happy to review whenever you're ready.

Reply:
should be ready now.